### PR TITLE
Add .editorconfig file with google-java-format compatible options

### LINKED ISSUE #1126: Add .editorconfig file
It is not easy for new contributors to ensure that their IDE is configured accordingly to the required code style. The only way right now is to carefully read Google Java Style Guide and change their IDE configuration manually. Or to remember to run `gradlew goJF` before commit.

The easier approach would be to provide [.editorconfig](https://editorconfig.org) file which would apply desired formatting rules automatically.

### PR DESCRIPTION
**Pull request by iNikem**
Fixes #1126 

Hopefully improved a little bit corresponding section in `CONTRIBUTING.md` as well.

Should we add git pre-commit hook as well into this repo? Or just add a corresponding instruction for users into `CONTRIBUTING.md`?

### CODE REVIEW

**Comment by bogdandrutu** [comment]
@iNikem we already have `./gradlew verJF` in the presubmit.

**Comment by iNikem** [comment]
@bogdandrutu I don't see that in my local repo's `.git/hooks/` folder. Where is it located?

**Comment by bogdandrutu** [comment]
it is not in git hooks, it is in PR hooks so CI will fail if not correctly formatted. Having in git hooks may look reasonable.

**Comment by iNikem** [comment]
I see. Yes, I meant adding `googleJavaFormat` not `verifyGoogleJavaFormat` into `pre-commit` hook so that wrong format couldn't leave developer's machine :)

**Comment by jkwatson** [comment]
I think adding a pre-commit hook script with instructions in the CONTRIBUTING doc about how to install it would be great.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -16,20 +16,30 @@ automatic check once you submit a PR, but you can also sign it after opening you
...
+or on Windows
 
-### Windows
+`gradlew.bat verifyGoogleJavaFormat`
 
-`gradlew.bat goJF`
+Instead of fixing style inconsistencies by hand, you can run gradle task `googleJavaFormat`
+to automatically fix all found issues:
+
+`./gradlew googleJavaFormat`
```
Comment (bogdandrutu) [inline CONTRIBUTING.md:34]: `goJF` shortcut

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -16,20 +16,30 @@ automatic check once you submit a PR, but you can also sign it after opening you
...
+which uses [google-java-format](https://github.com/google/google-java-format) library:
 
-### OS X or Linux
+`./gradlew verifyGoogleJavaFormat`
 
-`./gradlew goJF`
+or on Windows
 
-### Windows
+`gradlew.bat verifyGoogleJavaFormat`
```
Comment (bogdandrutu) [inline CONTRIBUTING.md:29]: `verJF`

File: CONTRIBUTING.md (hunk 1 above)
Comment (iNikem) [reply CONTRIBUTING.md:34]: Are you sure about that? I would prefer to have full names in official documentation. Makes discovery easier for newcomers. With abbreviation it is harder to Ctrl+F your way around.

Comment (jkwatson) [reply CONTRIBUTING.md:34]: I also prefer the full name in the documentation.

**Comment by iNikem** [comment]
Added an example of pre-commit hook.

**Comment by iNikem** [comment]
@bogdandrutu can you please merge this PR? Or do I need more reviewers?

**Pull request merged**